### PR TITLE
[Messenger] Extend SQS visibility timeout for messages that are still being processed

### PR DESCRIPTION
**Pull request by valtzu**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| License       | MIT

Now that https://github.com/symfony/symfony/pull/53508 is merged, let's add keepalive implementation for SQS too.

### CODE REVIEW

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/AmazonSqsIntegrationTest.php [hunk 1]
```diff
@@ -55,6 +56,20 @@ private function execute(string $dsn): void
...
         $this->assertEquals(['type' => DummyMessage::class, DummyMessage::class => 'special'], $encoded['headers']);
+
+        $this->waitUntilElapsed(seconds: 1.0, since: $messageSentAt);
+        $connection->keepalive($encoded['id']);
+        $this->waitUntilElapsed(seconds: 2.0, since: $messageSentAt);
+        $this->assertSame(0, $connection->getMessageCount(), 'The queue should be empty since visibility timeout was extended');
+        $connection->delete($encoded['id']);
+    }
+
+    private function waitUntilElapsed(float $seconds, float $since): void
```
Comment (OskarStark) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/AmazonSqsIntegrationTest.php:67]: What about using `symfony/clock`? IMHO this should be doable and ease testing a lot

Comment (nicolas-grekas) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/AmazonSqsIntegrationTest.php:67]: I'm not sure this makes sense: here we're doing an integration test, which means we'd need the remote server to also use a mock clock. That's not possible.

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php [hunk 2]
```diff
@@ -304,6 +304,19 @@ public function delete(string $id): void
...
+    public function keepalive(string $id): void
+    {
+        if (null === $this->configuration['visibility_timeout']) {
+            return;
+        }
+
+        $this->client->changeMessageVisibility([
+            'QueueUrl' => $this->getQueueUrl(),
+            'ReceiptHandle' => $id,
+            'VisibilityTimeout' => $this->configuration['visibility_timeout'],
```
Comment (valtzu) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:320]: I'm not sure if we should use `visibility_timeout` here or would it be better to use the alarm interval and some fixed multiplier, f.e. `$alarmInterval * 1.2` :thinking:

Or if we use `visibility_timeout`, then maybe we could warn if `visibility_timeout` is smaller than alarm interval.

What do you think?

Comment (fabpot) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:320]: Let's indeed check that `visibility_timeout` is smaller than the alarm interval.

Comment (valtzu) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:320]: The cleanest I can make this happen is to add `?int $seconds = null` parameter to every `keepalive` method in the chain. It makes a lot of sense to tell the transport for how long at minimum the message should be kept alive – but it means I have to touch `keepalive()` in `Worker`, `KeepaliveReceiverInterface`, `BeanstalkdReceiver` & `BeanstalkdTransport`  too. Also need to add `Application#getAlarmInterval()`.

The changes [are ready](https://github.com/symfony/symfony/commit/77008e38044e94e167953a903e749309555463ef), but I think it's better to do that in separate PR as it touches so many places. I'm proposing to merge this as-is after all and I'll open a new one for this keepalive ttl where we can discuss more if needed.

Comment (fabpot) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:320]: Can you please open the PR as it's a prerequisite for this one?

Comment (valtzu) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:320]: Sure, here: #58552

Comment (fabpot) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:320]: Now that #58552 is merged, can you rebase this one? Thank you.

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php [hunk 3]
```diff
@@ -26,14 +26,14 @@
...
-class AmazonSqsTransport implements TransportInterface, SetupableTransportInterface, MessageCountAwareInterface, ResetInterface
+class AmazonSqsTransport implements TransportInterface, KeepaliveReceiverInterface, SetupableTransportInterface, MessageCountAwareInterface, ResetInterface
 {
     private SerializerInterface $serializer;
 
     public function __construct(
         private Connection $connection,
         ?SerializerInterface $serializer = null,
-        private (ReceiverInterface&MessageCountAwareInterface)|null $receiver = null,
+        private (KeepaliveReceiverInterface&MessageCountAwareInterface)|null $receiver = null,
```
Comment (stof) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php]: this change making the constructor argument type stricter than before is a BC break.

Comment (valtzu) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php]: Right, reverted

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @valtzu.

**Pull request merged**